apps/web/src/components/GalleryEditor/GalleryOnboardingGuide/OnboardingDialogContext.tsx

```diff
@@ -0,0 +1,117 @@
+import { createContext, useCallback, useContext, useMemo, useState } from 'react';
+import { graphql, useFragment } from 'react-relay';
+
+import { OnboardingDialogContextFragment$key } from '~/generated/OnboardingDialogContextFragment.graphql';
+
+import useUpdateUserExperience from './useUpdateUserExperience';
+
+type OnboardingDialogContextType = {
+  step: number;
+  dialogMessage: string;
+  nextStep: () => void;
+  handleClose: () => void;
+};
+
+const OnboardingDialogMessage = {
+  1: 'You can modify your gallery name and description here.',
+  2: 'You can display multiple collections in a gallery. Name your first collection title here, and add a description.',
+  3: 'Search or filter through various chains and wallets to find the pieces you want to curate',
+  4: 'Click a piece to add to your collection.',
+  5: 'You can add multiple sections to adjust the number of columns and display your pieces in creative ways',
+};
+
+export const OnboardingDialogContext = createContext<OnboardingDialogContextType | undefined>(
+  undefined
+);
+
+type OnboardingDialogProviderProps = {
+  children: React.ReactNode;
+  queryRef: OnboardingDialogContextFragment$key;
+};
+
+export function OnboardingDialogProvider({ children, queryRef }: OnboardingDialogProviderProps) {
+  const query = useFragment(
+    graphql`
+      fragment OnboardingDialogContextFragment on Query {
+        viewer {
+          ... on Viewer {
+            __typename
+            userExperiences @required(action: THROW) {
+              type
+              experienced
+            }
+          }
+        }
+      }
+    `,
+    queryRef
+  );
+
+  const isUserExperiencedTooltips = useMemo(() => {
+    if (query.viewer?.__typename !== 'Viewer') return true;
+
+    return (
+      (query.viewer?.userExperiences.find(
+        (userExperience) => userExperience.type === 'MultiGalleryAnnouncement'
+      )?.experienced as boolean) || false
+    );
+  }, [query.viewer]);
+
+  const [step, setStep] = useState(1);
+
+  const updateUserExperience = useUpdateUserExperience();
+
+  const dialogMessage = useMemo(
+    () => OnboardingDialogMessage[step as keyof typeof OnboardingDialogMessage],
+    [step]
+  );
+
+  const dismissUserExperience = useCallback(async () => {
+    setStep(0);
```

Comment:
do we need to reset to step 0? not sure if this line is necessary

Reply:
Its kind of like hacky trick to dismiss the popup immediately. 

If not, there is some delay to dismiss since we need to wait for the API request to finish